### PR TITLE
Add force openSSL version with -d:sslVersion=x.x.x

### PR DESCRIPTION
**Pull request by treeform**
## General issue:

Other dynamic libraries (like libpg) load diffetent openSSL version then what nim loads.
Having two different openSSL loaded version causes a crash.
Use this compile time define to force the openSSL version that your other dynamic libraries use.

See: #10281, #10230, https://github.com/nim-lang/Nim/issues/9419

Use on command line:
```
nim c -r -d:ssl -d:sslVersion=1.0.0 foo.nim
```

Or in nim.cfg
```
define:"sslVersion=1.0.0"
```

### CODE REVIEW

File: lib/wrappers/openssl.nim [hunk 1]
```diff
@@ -26,7 +26,31 @@
 
 const useWinVersion = defined(Windows) or defined(nimdoc)
 
-when useWinVersion:
+## To force openSSL version use -d:sslVersion=1.0.0
+## See: #10281, #10230
+## General issue:
+## Other dynamic libraries (like libpg) load diffetent openSSL version then what nim loads.
+## Having two different openSSL loaded version causes a crash.
+## Use this compile time define to force the openSSL version that your other dynamic libraries want.
```
Comment (dom96) [inline lib/wrappers/openssl.nim]: Note that the doc gen will show this as one big paragraph which likely isn't your intention.

Comment (treeform) [reply lib/wrappers/openssl.nim]: You right. I will change it to a single comment so that it does not show up in docs.

Comment (dom96) [reply lib/wrappers/openssl.nim]: Well, it would be nice to have this somewhere in our docs :)

**Comment by treeform** [comment]
If users has random SSL versions installed. Your program will crash. Because one program would load 2 different versions of SSL DLLs. At-least with this method the program has a chance of running if user has the right DLLs installed. I would recommend shipping the right DLLs with your program. And this is what this is about forcing the program to load the DLLs you specify give it rather then some random DLLs which cause it to crash.

You can tell which libpg loads by using otool:
```
otool -L /usr/local/Cellar/libpq/10.5/lib/libpq.5.10.dylib
/usr/local/Cellar/libpq/10.5/lib/libpq.5.10.dylib:
        /usr/local/opt/libpq/lib/libpq.5.dylib (compatibility version 5.0.0, current version 5.10.0)
        /usr/local/opt/openssl/lib/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.50.4)
```

**Comment by dom96** [comment]
This is still a solution that requires the programmer to be aware of it though. Do you know if we could copy how libpg determines which SSL library it loads?

**Comment by treeform** [comment]
I did have a simpler solution: https://github.com/nim-lang/Nim/pull/10230

But it was reverted bc of another bug (Nim claiming to support ssl it
doesn't, or ssl user has install is not compatible)

It's a huge mess...

**Comment by Araq** [comment]
Definitely an improvement over the status quo, merging.

**Pull request merged**